### DISCUSSION TITLE
Stata Error : Invalid File specification (r198) ------

### DISCUSSION THREAD

**Discussion by bnhippolyte** [gui2de/ppol768-spring23#170] [Q&A] [answered]
Hi all,

I was wondering if anyone encountered this error when making changes to their do file .. 

After making changes to my do file, I typed the code *save, replace* and received an error from Stata **" Invalid file specification "**... If anyone knows how I could get around this error, please help. 

It's question 4. 

Thanks. (Link to my code below) 

Individual Assignments/Hippolyte BeverlyAnn/week-04/assignment1_.do

**Comment by hamzamajoka** [comment, marked as answer]
Thank you for posting this here.

There are a couple of issues:

1) It's not a good idea to use "save, replace" command. The main idea behind this is that you should have distinct input and output files (see this article from week 2 readings: https://web.stanford.edu/~gentzkow/research/CodeAndData.pdf). If you overwrite your original dataset, you won't be able to re-run your code. For example, in question 1, you are generating the following variable: pixel_min. If you use the "save, replace" command in your do file, your original dataset will get over-written with the dataset containing these new variables. And when you try to re-run your do file, you will get an error when generating pixel_min variable (because it already exists in the dataset) This defeats the main purpose of writing your code in a do file (replication!) 

2) The error is unusual, this is the first time I've seen this error. I looked into it and apparently you can't use "save, replace" command if you have changed the structure of the dataset (e.g. if you use reshape command) That's why "save, replace" command might have worked in the first 3 questions where you were just adding new columns, but didn't work in q4 where you reshape the data.

3) If you HAVE to overwrite a dataset then use the following syntax: "save name_of_the_dataset, replace" It reduces the possibility of mistakenly over-writing your dataset.
